lib/std/fs.zig

```diff
@@ -574,10 +593,14 @@ pub const Dir = struct {
             pub fn next(self: *Self) Error!?Entry {
                 start_over: while (true) {
                     if (self.index >= self.end_index) {
+                        if (self.first_iter) {
+                            std.os.lseek_SET(self.dir.fd, 0) catch unreachable; // Can't fail
```

Comment:
This should probably have the `// EBADF here likely means that the Dir was not opened with iteration permissions` comment as well